tests/D2L.CodeStyle.Analyzers.Test/Specs/DependencyRegistrationsAnalyzer.cs

```diff
@@ -312,19 +313,23 @@ public sealed class ImmutableThingFactory :
 		IFactory<ImmutableThing>, 
 		IFactory<ImmutableThing, Type> { }
 
+	[Objects.Immutable]
 	internal sealed class ThingWithInternalConstructor : IImmutableSingleton {

```

Comment:
`GetImmutabilityScope` did one level of recursion (it would check if any of your interfaces had `[Immutable]`), so switching to `ImmutabilityContext.IsImmutable` requires this.

Let's standardize on the smaller set of rules here.